[CI] SslMultiPortTests.testThatProfileTransportClientCanConnectToNoClientAuthProfile failure in master

Console log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=sles/183/console

```
REPRODUCE WITH: ./gradlew :x-pack:plugin:security:unitTest \
  -Dtests.seed=5485428EAADFA85E \
  -Dtests.class=org.elasticsearch.xpack.security.transport.ssl.SslMultiPortTests \
  -Dtests.method="testThatProfileTransportClientCanConnectToNoClientAuthProfile" \
  -Dtests.security.manager=true \
  -Dtests.locale=fr-BE \
  -Dtests.timezone=America/Mexico_City \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Errors:

```
15:15:17   1> [2019-01-15T07:15:09,768][WARN ][o.e.t.TcpTransport       ] [node_sc2] exception caught on transport layer [TcpNioSocketChannel{localAddress=/127.0.0.1:39091, remoteAddress=/127.0.0.1:38608}], closing connection
15:15:17   1> javax.net.ssl.SSLHandshakeException: null cert chain
15:15:17   1> 	at sun.security.ssl.Handshaker.checkThrown(Handshaker.java:1521) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.checkTaskThrown(SSLEngineImpl.java:528) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.writeAppRecord(SSLEngineImpl.java:1197) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.wrap(SSLEngineImpl.java:1165) ~[?:?]
15:15:17   1> 	at javax.net.ssl.SSLEngine.wrap(SSLEngine.java:509) ~[?:1.8.0_192]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver.wrap(SSLDriver.java:218) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver.access$1000(SSLDriver.java:51) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.handshake(SSLDriver.java:350) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.read(SSLDriver.java:382) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver.read(SSLDriver.java:123) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLChannelContext.read(SSLChannelContext.java:143) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.nio.EventHandler.handleRead(EventHandler.java:119) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.handleRead(NioSelector.java:398) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.processKey(NioSelector.java:246) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:172) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:129) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
15:15:17   1> Caused by: javax.net.ssl.SSLHandshakeException: null cert chain
15:15:17   1> 	at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1647) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:318) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:306) ~[?:?]
15:15:17   1> 	at sun.security.ssl.ServerHandshaker.clientCertificate(ServerHandshaker.java:1939) ~[?:?]
15:15:17   1> 	at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:232) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1037) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker$1.run(Handshaker.java:970) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker$1.run(Handshaker.java:967) ~[?:?]
15:15:17   1> 	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_192]
15:15:17   1> 	at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1459) ~[?:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.runTasks(SSLDriver.java:433) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.handshake(SSLDriver.java:358) ~[main/:?]
15:15:17   1> 	... 9 more
15:15:17   1> [2019-01-15T07:15:09,769][WARN ][o.e.t.TcpTransport       ] [_client_] send message failed [channel: Netty4TcpChannel{localAddress=0.0.0.0/0.0.0.0:38608, remoteAddress=/127.0.0.1:39091}]
15:15:17   1> javax.net.ssl.SSLException: Received fatal alert: bad_certificate
15:15:17   1> 	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1647) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1615) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.recvAlert(SSLEngineImpl.java:1781) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:1070) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:896) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:766) ~[?:?]
15:15:17   1> 	at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624) ~[?:1.8.0_192]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295) ~[netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1301) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1203) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1247) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:656) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:556) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:510) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:470) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:909) [netty-common-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
```

```
15:15:17   1> [2019-01-15T07:15:10,853][WARN ][o.e.t.TcpTransport       ] [node_sc2] exception caught on transport layer [TcpNioSocketChannel{localAddress=/0:0:0:0:0:0:0:1:45775, remoteAddress=/0:0:0:0:0:0:0:1:52100}], closing connection
15:15:17   1> javax.net.ssl.SSLHandshakeException: null cert chain
15:15:17   1> 	at sun.security.ssl.Handshaker.checkThrown(Handshaker.java:1521) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.checkTaskThrown(SSLEngineImpl.java:528) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.writeAppRecord(SSLEngineImpl.java:1197) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.wrap(SSLEngineImpl.java:1165) ~[?:?]
15:15:17   1> 	at javax.net.ssl.SSLEngine.wrap(SSLEngine.java:509) ~[?:1.8.0_192]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver.wrap(SSLDriver.java:218) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver.access$1000(SSLDriver.java:51) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.handshake(SSLDriver.java:350) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.read(SSLDriver.java:382) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver.read(SSLDriver.java:123) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLChannelContext.read(SSLChannelContext.java:143) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.nio.EventHandler.handleRead(EventHandler.java:119) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.handleRead(NioSelector.java:398) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.processKey(NioSelector.java:246) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:172) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:129) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
15:15:17   1> Caused by: javax.net.ssl.SSLHandshakeException: null cert chain
15:15:17   1> 	at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1647) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:318) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:306) ~[?:?]
15:15:17   1> 	at sun.security.ssl.ServerHandshaker.clientCertificate(ServerHandshaker.java:1939) ~[?:?]
15:15:17   1> 	at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:232) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1037) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker$1.run(Handshaker.java:970) ~[?:?]
15:15:17   1> 	at sun.security.ssl.Handshaker$1.run(Handshaker.java:967) ~[?:?]
15:15:17   1> 	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_192]
15:15:17   1> 	at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1459) ~[?:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.runTasks(SSLDriver.java:433) ~[main/:?]
15:15:17   1> 	at org.elasticsearch.xpack.security.transport.nio.SSLDriver$HandshakeMode.handshake(SSLDriver.java:358) ~[main/:?]
15:15:17   1> 	... 9 more
15:15:17   1> [2019-01-15T07:15:10,854][WARN ][o.e.t.TcpTransport       ] [programmatic_transport_client] send message failed [channel: Netty4TcpChannel{localAddress=0.0.0.0/0.0.0.0:52100, remoteAddress=/0:0:0:0:0:0:0:1:45775}]
15:15:17   1> javax.net.ssl.SSLException: Received fatal alert: bad_certificate
15:15:17   1> 	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1647) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1615) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.recvAlert(SSLEngineImpl.java:1781) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:1070) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:896) ~[?:?]
15:15:17   1> 	at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:766) ~[?:?]
15:15:17   1> 	at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624) ~[?:1.8.0_192]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295) ~[netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1301) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1203) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1247) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:656) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:556) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:510) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:470) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:909) [netty-common-4.1.32.Final.jar:4.1.32.Final]
15:15:17   1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
```

```
15:15:17   1> [2019-01-15T07:15:15,612][INFO ][o.e.x.s.t.s.SslMultiPortTests] [testThatProfileTransportClientCanConnectToNoClientAuthProfile] after test
15:15:17 ERROR   1.01s J2 | SslMultiPortTests.testThatProfileTransportClientCanConnectToNoClientAuthProfile <<< FAILURES!
15:15:17    > Throwable #1: NoNodeAvailableException[None of the configured nodes are available: [{#transport#-1}{y0eeGtT0Q_-mZrpu-6h6qg}{localhost}{127.0.0.1:49750}]]
15:15:17    > 	at __randomizedtesting.SeedInfo.seed([5485428EAADFA85E:8149D04C22ACA7AD]:0)
15:15:17    > 	at org.elasticsearch.client.transport.TransportClientNodesService.ensureNodesAreAvailable(TransportClientNodesService.java:351)
15:15:17    > 	at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:249)
15:15:17    > 	at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:57)
15:15:17    > 	at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:386)
15:15:17    > 	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:393)
15:15:17    > 	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:382)
15:15:17    > 	at org.elasticsearch.client.support.AbstractClient$ClusterAdmin.execute(AbstractClient.java:676)
15:15:17    > 	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:45)
15:15:17    > 	at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:52)
15:15:17    > 	at org.elasticsearch.test.SecurityIntegTestCase.assertGreenClusterState(SecurityIntegTestCase.java:410)
15:15:17    > 	at org.elasticsearch.xpack.security.transport.ssl.SslMultiPortTests.testThatProfileTransportClientCanConnectToNoClientAuthProfile(SslMultiPortTests.java:179)
15:15:17    > 	at java.lang.Thread.run(Thread.java:748)
15:15:17   1> [2019-01-15T07:15:15,631][INFO ][o.e.n.Node               ] [suite] stopping ...
15:15:17   1> [2019-01-15T07:15:15,635][INFO ][o.e.c.s.ClusterApplierService] [node_sc2] master node changed {previous [{node_s0}{uKqyec8oRTea2qrzLK1cEA}{ROSCEfbKSwCU0UDg1U19mw}{127.0.0.1}{127.0.0.1:37389}{xpack.installed=true}], current []}, term: 1, version: 97, reason: becoming candidate: onLeaderFailure
15:15:17   1> [2019-01-15T07:15:15,635][INFO ][o.e.c.s.ClusterApplierService] [node_s1] master node changed {previous [{node_s0}{uKqyec8oRTea2qrzLK1cEA}{ROSCEfbKSwCU0UDg1U19mw}{127.0.0.1}{127.0.0.1:37389}{xpack.installed=true}], current []}, term: 1, version: 97, reason: becoming candidate: onLeaderFailure
15:15:17   1> [2019-01-15T07:15:15,638][WARN ][o.e.c.NodeConnectionsService] [node_sc2] failed to connect to node {node_s0}{uKqyec8oRTea2qrzLK1cEA}{ROSCEfbKSwCU0UDg1U19mw}{127.0.0.1}{127.0.0.1:37389}{xpack.installed=true} (tried [1] times)
15:15:17   1> org.elasticsearch.transport.ConnectTransportException: [node_s0][127.0.0.1:37389] connect_exception
15:15:17   1> 	at org.elasticsearch.transport.TcpTransport$ChannelsConnectedListener.onFailure(TcpTransport.java:1566) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.action.ActionListener.lambda$toBiConsumer$2(ActionListener.java:99) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.common.concurrent.CompletableContext.lambda$addListener$0(CompletableContext.java:42) ~[elasticsearch-core-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760) ~[?:1.8.0_192]
15:15:17   1> 	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:778) ~[?:1.8.0_192]
15:15:17   1> 	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2140) ~[?:1.8.0_192]
15:15:17   1> 	at org.elasticsearch.common.concurrent.CompletableContext.addListener(CompletableContext.java:45) ~[elasticsearch-core-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.SocketChannelContext.addConnectListener(SocketChannelContext.java:82) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSocketChannel.addConnectListener(NioSocketChannel.java:77) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.nio.NioTcpChannel.addConnectListener(NioTcpChannel.java:62) ~[transport-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TcpTransport.initiateConnection(TcpTransport.java:337) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TcpTransport.openConnection(TcpTransport.java:304) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.ConnectionManager.internalOpenConnection(ConnectionManager.java:223) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.ConnectionManager.connectToNode(ConnectionManager.java:109) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:344) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:331) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.cluster.NodeConnectionsService.validateAndConnectIfNeeded(NodeConnectionsService.java:156) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.cluster.NodeConnectionsService$1.doRun(NodeConnectionsService.java:109) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:759) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_192]
15:15:17   1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_192]
15:15:17   1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
15:15:17   1> Caused by: java.net.ConnectException: Connection refused
15:15:17   1> 	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
15:15:17   1> 	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[?:?]
15:15:17   1> 	at org.elasticsearch.nio.SocketChannelContext.connect(SocketChannelContext.java:116) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.EventHandler.handleConnect(EventHandler.java:97) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.attemptConnect(NioSelector.java:406) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.setUpNewChannels(NioSelector.java:423) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.preSelect(NioSelector.java:260) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:153) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:129) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	... 1 more
15:15:17   1> [2019-01-15T07:15:15,641][INFO ][o.e.n.Node               ] [suite] stopped
15:15:17   1> [2019-01-15T07:15:15,641][INFO ][o.e.n.Node               ] [suite] closing ...
15:15:17   1> [2019-01-15T07:15:15,640][WARN ][o.e.c.NodeConnectionsService] [node_s1] failed to connect to node {node_s0}{uKqyec8oRTea2qrzLK1cEA}{ROSCEfbKSwCU0UDg1U19mw}{127.0.0.1}{127.0.0.1:37389}{xpack.installed=true} (tried [1] times)
15:15:17   1> org.elasticsearch.transport.ConnectTransportException: [node_s0][127.0.0.1:37389] connect_exception
15:15:17   1> 	at org.elasticsearch.transport.TcpTransport$ChannelsConnectedListener.onFailure(TcpTransport.java:1566) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.action.ActionListener.lambda$toBiConsumer$2(ActionListener.java:99) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.common.concurrent.CompletableContext.lambda$addListener$0(CompletableContext.java:42) ~[elasticsearch-core-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760) ~[?:1.8.0_192]
15:15:17   1> 	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:778) ~[?:1.8.0_192]
15:15:17   1> 	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2140) ~[?:1.8.0_192]
15:15:17   1> 	at org.elasticsearch.common.concurrent.CompletableContext.addListener(CompletableContext.java:45) ~[elasticsearch-core-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.SocketChannelContext.addConnectListener(SocketChannelContext.java:82) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSocketChannel.addConnectListener(NioSocketChannel.java:77) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.nio.NioTcpChannel.addConnectListener(NioTcpChannel.java:62) ~[transport-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TcpTransport.initiateConnection(TcpTransport.java:337) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TcpTransport.openConnection(TcpTransport.java:304) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.ConnectionManager.internalOpenConnection(ConnectionManager.java:223) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.ConnectionManager.connectToNode(ConnectionManager.java:109) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:344) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:331) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.cluster.NodeConnectionsService.validateAndConnectIfNeeded(NodeConnectionsService.java:156) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.cluster.NodeConnectionsService$1.doRun(NodeConnectionsService.java:109) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:759) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_192]
15:15:17   1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_192]
15:15:17   1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
15:15:17   1> Caused by: java.net.ConnectException: Connection refused
15:15:17   1> 	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
15:15:17   1> 	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[?:?]
15:15:17   1> 	at org.elasticsearch.nio.SocketChannelContext.connect(SocketChannelContext.java:116) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.EventHandler.handleConnect(EventHandler.java:97) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.attemptConnect(NioSelector.java:406) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.setUpNewChannels(NioSelector.java:423) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.preSelect(NioSelector.java:260) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:153) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:129) ~[elasticsearch-nio-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
15:15:17   1> 	... 1 more
```

Didn't manage to reproduce locally.

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ssl/SslMultiPortTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ssl/SslMultiPortTests.java
@@ -7,6 +7,7 @@ package org.elasticsearch.xpack.security.transport.ssl;
 
 import org.elasticsearch.client.transport.NoNodeAvailableException;
 import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.network.NetworkAddress;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.transport.TransportAddress;
 import org.elasticsearch.test.SecurityIntegTestCase;
@@ -34,11 +35,13 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
 
     private static int randomClientPort;
     private static int randomNoClientAuthPort;
+    private static InetAddress localAddress;
 
     @BeforeClass
     public static void getRandomPort() {
         randomClientPort = randomIntBetween(49000, 65500); // ephemeral port
         randomNoClientAuthPort = randomIntBetween(49000, 65500);
+        localAddress = InetAddress.getLoopbackAddress();
     }
 
     /**
@@ -65,12 +68,11 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
         Settings.Builder builder = Settings.builder().put(super.nodeSettings(nodeOrdinal));
         addSSLSettingsForNodePEMFiles(builder, "transport.profiles.client.xpack.security.", true);
         builder.put("transport.profiles.client.port", randomClientPortRange)
-            // make sure this is "localhost", no matter if ipv4 or ipv6, but be consistent
-            .put("transport.profiles.client.bind_host", "localhost")
+            .put("transport.profiles.client.bind_host", NetworkAddress.format(localAddress))
             .put("transport.profiles.client.xpack.security.ssl.certificate_authorities", trustCert.toAbsolutePath());
         addSSLSettingsForNodePEMFiles(builder, "transport.profiles.no_client_auth.xpack.security.", true);
         builder.put("transport.profiles.no_client_auth.port", randomNoClientAuthPortRange)
-            .put("transport.profiles.no_client_auth.bind_host", "localhost")
+            .put("transport.profiles.no_client_auth.bind_host", NetworkAddress.format(localAddress))
             .put("transport.profiles.no_client_auth.xpack.security.ssl.client_authentication", SSLClientAuth.NONE);
         final Settings settings = builder.build();
         logger.info("node {} settings:\n{}", nodeOrdinal, settings);
@@ -117,7 +119,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
                 .put("node.name", "programmatic_transport_client")
                 .put("cluster.name", internalCluster().getClusterName())
                 .build(), LocalStateSecurity.class)) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(),
+            transportClient.addTransportAddress(new TransportAddress(localAddress,
                     getProfilePort("no_client_auth")));
             assertGreenClusterState(transportClient);
         }
@@ -132,7 +134,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
      */
     public void testThatStandardTransportClientCannotConnectToClientProfile() throws Exception {
         try (TransportClient transportClient = createTransportClient(Settings.EMPTY)) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(), getProfilePort("client")));
+            transportClient.addTransportAddress(new TransportAddress(localAddress, getProfilePort("client")));
             transportClient.admin().cluster().prepareHealth().get();
             fail("Expected NoNodeAvailableException");
         } catch (NoNodeAvailableException e) {
@@ -154,7 +156,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
             "/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testclient-client-profile.crt",
             Arrays.asList("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.crt"));
         try (TransportClient transportClient = createTransportClient(builder.build())) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(), getProfilePort("client")));
+            transportClient.addTransportAddress(new TransportAddress(localAddress, getProfilePort("client")));
             assertGreenClusterState(transportClient);
         }
     }
@@ -174,7 +176,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
             "/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testclient-client-profile.crt",
             Arrays.asList("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.crt"));
         try (TransportClient transportClient = createTransportClient(builder.build())) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(),
+            transportClient.addTransportAddress(new TransportAddress(localAddress,
                     getProfilePort("no_client_auth")));
             assertGreenClusterState(transportClient);
         }
@@ -234,7 +236,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
                 .build();
         try (TransportClient transportClient = new TestXPackTransportClient(settings,
                                                                             Collections.singletonList(LocalStateSecurity.class))) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(), getProfilePort("client")));
+            transportClient.addTransportAddress(new TransportAddress(localAddress, getProfilePort("client")));
             assertGreenClusterState(transportClient);
             fail("Expected NoNodeAvailableException");
         } catch (NoNodeAvailableException e) {
@@ -253,7 +255,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
                 .build();
         try (TransportClient transportClient = new TestXPackTransportClient(settings,
                                                                             Collections.singletonList(LocalStateSecurity.class))) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(),
+            transportClient.addTransportAddress(new TransportAddress(localAddress,
                     getProfilePort("no_client_auth")));
             assertGreenClusterState(transportClient);
             fail("Expected NoNodeAvailableException");
@@ -276,7 +278,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
             .build();
         try (TransportClient transportClient = new TestXPackTransportClient(settings,
                                                                             Collections.singletonList(LocalStateSecurity.class))) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(),
+            transportClient.addTransportAddress(new TransportAddress(localAddress,
                     getProfilePort("no_client_auth")));
         }
     }
@@ -297,7 +299,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
             .build();
         try (TransportClient transportClient = new TestXPackTransportClient(settings,
                                                                             Collections.singletonList(LocalStateSecurity.class))) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(), getProfilePort("client")));
+            transportClient.addTransportAddress(new TransportAddress(localAddress, getProfilePort("client")));
             assertGreenClusterState(transportClient);
             fail("Expected NoNodeAvailableException");
         } catch (NoNodeAvailableException e) {
@@ -365,7 +367,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
             .build();
         try (TransportClient transportClient = new TestXPackTransportClient(settings,
                                                                             Collections.singletonList(LocalStateSecurity.class))) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(), getProfilePort("client")));
+            transportClient.addTransportAddress(new TransportAddress(localAddress, getProfilePort("client")));
             assertGreenClusterState(transportClient);
             fail("Expected NoNodeAvailableException");
         } catch (NoNodeAvailableException e) {
@@ -387,7 +389,7 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
             .build();
         try (TransportClient transportClient = new TestXPackTransportClient(settings,
                                                                             Collections.singletonList(LocalStateSecurity.class))) {
-            transportClient.addTransportAddress(new TransportAddress(InetAddress.getLoopbackAddress(),
+            transportClient.addTransportAddress(new TransportAddress(localAddress,
                     getProfilePort("no_client_auth")));
             assertGreenClusterState(transportClient);
             fail("Expected NoNodeAvailableException");
@@ -397,8 +399,14 @@ public class SslMultiPortTests extends SecurityIntegTestCase {
     }
 
     private static int getProfilePort(String profile) {
-        TransportAddress transportAddress =
-                randomFrom(internalCluster().getInstance(Transport.class).profileBoundAddresses().get(profile).boundAddresses());
-        return transportAddress.address().getPort();
+        TransportAddress[] transportAddresses =
+            internalCluster().getInstance(Transport.class).profileBoundAddresses().get(profile).boundAddresses();
+        for (TransportAddress address : transportAddresses) {
+            if (address.address().getAddress().equals(localAddress)) {
+                return address.address().getPort();
+            }
+        }
+        throw new IllegalStateException("failed to find transport address equal to [" + NetworkAddress.format(localAddress) + "] " +
+            " in the following bound addresses " + Arrays.toString(transportAddresses));
     }
 }

